### PR TITLE
Bazel 3.7.2

### PR DESCRIPTION
**Pull request by aherrmann**
Update Bazel from version 3.3.1 to 3.7.2.
Updates the nixpkgs revision accordingly.
Updates rules_haskell dependencies to latest released versions.

### CODE REVIEW

File: shell.nix [hunk 1]
```diff
@@ -18,6 +18,7 @@ mkShell {
     perl
     python3
     bazel
+    jdk11
```
Comment (aherrmann) [inline shell.nix:21]: This is added to avoid [this error](https://github.com/tweag/rules_haskell/runs/1864772907#step:7:380):
```
       ERROR: /home/runner/.cache/bazel/_bazel_runner/c91220d87f26118b19246641d5266d21/external/local_jdk/BUILD.bazel:3:10: in fail_rule rule @local_jdk//:jdk: 
       Traceback (most recent call last):
       	File "/home/runner/.cache/bazel/_bazel_runner/c91220d87f26118b19246641d5266d21/external/bazel_tools/tools/jdk/fail_rule.bzl", line 19, column 13, in _fail_rule_impl
       		fail("%s %s" % (ctx.attr.header, ctx.attr.message))
       Error in fail: Auto-Configuration Error: Cannot find Java binary bin/java in /home/runner/.cache/bazel/_bazel_runner/install/a704d98024e76cfbf273992773847901/embedded_tools/tools/jdk/nosystemjdk; either correct your JAVA_HOME, PATH or specify embedded Java (e.g. --javabase=@bazel_tools//tools/jdk:remote_jdk11)
       ERROR: Analysis of target '//debug/linking_utils:test-ldd' failed; build aborted: Analysis of target '@local_jdk//:jdk' failed
       INFO: Elapsed time: 1.937s
       INFO: 0 processes.
       FAILED: Build did NOT complete successfully (0 packages loaded, 304 targets configured)
       ERROR: Couldn't start the build. Unable to run tests
       FAILED: Build did NOT complete successfully (0 packages loaded, 304 targets configured)
       INFO: Streaming build results to: https://app.buildbuddy.io/invocation/2c151cbc-1caa-483a-835e-ba30de50c5ff
       FAILED: Build did NOT complete successfully (0 packages loaded, 304 targets configured)
```
A better solution would be a rules_nixpkgs provided Java toolchain, but that is out of scope for this PR.

**Comment by facundominguez** [comment]
The changes themselves look ok. I tried `test //...` in centos 7 and got:
```
$ nix-shell --pure --run "bazel test //..."
DEBUG: /home/centos/rules_haskell/vendor/bazel_json/lib/json_rules.bzl:180:14: defined null rule
DEBUG: /home/centos/rules_haskell/vendor/bazel_json/lib/json_rules.bzl:183:14: defined string rule
DEBUG: /home/centos/rules_haskell/vendor/bazel_json/lib/json_rules.bzl:186:14: defined number rule
DEBUG: /home/centos/rules_haskell/vendor/bazel_json/lib/json_rules.bzl:189:14: got a list
DEBUG: /home/centos/rules_haskell/vendor/bazel_json/lib/json_rules.bzl:191:14: got a dict
WARNING: /home/centos/rules_haskell/haskell/platforms/BUILD.bazel:26:6: target '//haskell/platforms:darwin_x86_64_nixpkgs' is deprecated: Use @io_tweag_rules_nixpkgs//nixpkgs/platforms:host instead.
WARNING: /home/centos/rules_haskell/haskell/platforms/BUILD.bazel:19:6: target '//haskell/platforms:linux_x86_64_nixpkgs' is deprecated: Use @io_tweag_rules_nixpkgs//nixpkgs/platforms:host instead.
DEBUG: /home/centos/rules_haskell/vendor/bazel_json/lib/json_rules.bzl:167:14: JSON(//vendor/bazel_json/test:null_check): None
DEBUG: /home/centos/rules_haskell/vendor/bazel_json/lib/json_rules.bzl:167:14: JSON(//vendor/bazel_json/test:boolean_check): False
DEBUG: /home/centos/rules_haskell/vendor/bazel_json/lib/json_rules.bzl:167:14: JSON(//vendor/bazel_json/test:string_check): 
DEBUG: /home/centos/rules_haskell/vendor/bazel_json/lib/json_rules.bzl:167:14: JSON(//vendor/bazel_json/test:string_check_quoted): my 'quoted' json_string
DEBUG: /home/centos/rules_haskell/vendor/bazel_json/lib/json_rules.bzl:167:14: JSON(//vendor/bazel_json/test:string_check_non_empty): my_json_string
DEBUG: /home/centos/rules_haskell/vendor/bazel_json/lib/json_rules.bzl:167:14: JSON(//vendor/bazel_json/test:number_check_77): (77, 0)
DEBUG: /home/centos/rules_haskell/vendor/bazel_json/lib/json_rules.bzl:167:14: JSON(//vendor/bazel_json/test:number_check_fraction): (77, -2)
DEBUG: /home/centos/rules_haskell/vendor/bazel_json/lib/json_rules.bzl:167:14: JSON(//vendor/bazel_json/test:array_check_values): [None, False, "", (0, 0), [], {}]
DEBUG: /home/centos/rules_haskell/vendor/bazel_json/lib/json_rules.bzl:167:14: JSON(//vendor/bazel_json/test:object_check_empty): {"an_array": [None, False, "", (0, 0), [], {}]}
ERROR: /home/centos/rules_haskell/tests/library-with-sysincludes/BUILD.bazel:20:11: in cc_library rule //tests/library-with-sysincludes:zlib: header 'external/zlib.dev/include/zconf.h' is not under the specified strip prefix 'external/zlib.dev/include'
ERROR: /home/centos/rules_haskell/tests/library-with-sysincludes/BUILD.bazel:20:11: in cc_library rule //tests/library-with-sysincludes:zlib: header 'external/zlib.dev/include/zlib.h' is not under the specified strip prefix 'external/zlib.dev/include'
ERROR: Analysis of target '//tests/library-with-sysincludes:zlib' failed; build aborted: Analysis of target '//tests/library-with-sysincludes:zlib' failed
INFO: Elapsed time: 1.848s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (173 packages loaded, 1290 targets configured)
FAILED: Build did NOT complete successfully (173 packages loaded, 1290 targets configured)
    currently loading: @stackage-exe//proto-lens-protoc ... (4 packages)
    Fetching @ghcide; fetching
    Fetching @stackage; Restarting.
    Fetching @nixpkgs_pandoc; Building Nix derivation
    Fetching @rules_haskell_ghc_nixpkgs_haskell_toolchain; Restarting.
    Fetching @rules_haskell_ghc_nixpkgs; Building Nix derivation
    Fetching @rules_haskell_sh_posix_nixpkgs; Building Nix derivation
```

**Comment by facundominguez** [comment]
That fixes it for me!

**Review by facundominguez — APPROVED** [review]

**Comment by aherrmann** [comment]
@Mergifyio rebase

**Comment by aherrmann** [comment]
@Mergifyio refresh
